### DISCUSSION TITLE
erasure set 12，parity 4，stop one minio node，it can't put object？

### DISCUSSION THREAD

**Discussion by lddlww** [minio/minio#14388] [Q&A] [answered]
minio version：
```
minio version RELEASE.2022-02-18T01-50-10Z
```
minio toplogy:
```
4 nodes, 24 drives per node
erasure set size: 12
parity: 4
```

when i stop one minio node, then use ```mc mb myminio/brtest```, but it throws follow error:
```
mcli: <ERROR> Unable to make bucket `myminio/brtest`. Resource requested is unreadable, please reduce your request rate
```

reference to minio calculator：https://min.io/product/erasure-code-calculator, this topology can tolerance 1 node or 4 drives (per erasure set) fails, but it seems not, when i stop one minio node, it can't write

did i miss something?

**Comment by klauspost** [comment]
You need  at least 1 extra parity to write. When you are at the "limit", we only allow reads.

I agree the calculator doesn't make that clear at all.

**Reply by lddlww** [reply]
thanks for reply

how can i accuracy to compute minio tolerance？

**Reply by klauspost** [reply]
Wait, do you have K+M=12 , or K=12 and M=4 (and K+M=16)?

Could you post the exact config you have, and an `mc admin info --json myminio`?

**Reply by lddlww** [reply]
reference follow
![image](https://user-images.githubusercontent.com/27006237/155319358-4ebb25cd-e90c-4d32-937f-e9e0e9a67c6a.png)

**Comment by klauspost** [comment]
@lddlww No, your setup that produces the error.

**Reply by lddlww** [reply]
but is auto generated by minio

**Reply by lddlww** [reply]
@klauspost  how can i set？

**Comment by lddlww** [comment]
one minio topology：
```
4 nodes, 1 drive per node
erasure set size: 4
parity: 2
```
when i shutdown one minio service, it can be write always

another minio topology:
```
4 nodes, 4 drives per node
erasure set size: 4
parity: 2
```
when i shutdown one minio service, it can't be write 

i am a bit confuse about above difference tolerance

**Reply by klauspost** [reply]
Please post the *exact setup* and mc info as requested above, on the setup that produces `mcli: <ERROR> Unable to make bucket `myminio/brtest`. Resource requested is unreadable, please reduce your request rate`.

**Reply by lddlww** [reply]
```Could you post the exact config you have, and an mc admin info --json myminio?```
[minio.info.gz](https://github.com/minio/minio/files/8124388/minio.info.gz)
this is myminio info

minio topology: (auto generate by minio)
```
4 nodes, 24 drives per node
erasure set size（K+M）: 12
parity（M）: 4
```

**Reply by lddlww** [reply]
when i set ```MINIO_ERASURE_SET_DRIVE_COUNT=16``` minio service can't start and throw follow error: 
```
 minio[1738291]: ERROR Invalid command line arguments: Invalid set drive count. Acceptable values for 24 number drives are [4 6 8 12]
```
seems maximum ```erasure set size``` is 12, but according minio calculator the maximum of ```erasure set size``` is 16

**Reply by lddlww** [reply]
mc admin info myminio
```
●  10.5.17.100:9022
   Uptime: 2 seconds 
   Version: 2022-02-18T01:50:10Z
   Network: 4/4 OK 
   Drives: 24/24 OK 
   Pool: 4th

●  10.5.17.97:9022
   Uptime: 18 minutes 
   Version: 2022-02-18T01:50:10Z
   Network: 4/4 OK 
   Drives: 24/24 OK 
   Pool: 1st

●  10.5.17.98:9022
   Uptime: 18 minutes 
   Version: 2022-02-18T01:50:10Z
   Network: 4/4 OK 
   Drives: 24/24 OK 
   Pool: 2nd

●  10.5.17.99:9022
   Uptime: 18 minutes 
   Version: 2022-02-18T01:50:10Z
   Network: 4/4 OK 
   Drives: 24/24 OK 
   Pool: 3rd
```
mc admin info --json myminio | jq '[.info.servers|.[]|.drives|.[]|.disk_index]|max+1'
```
12
```
mc admin info --json clickhouse | jq .info.backend
```
{
  "backendType": "Erasure",
  "onlineDisks": 96,
  "rrSCParity": 4,
  "standardSCParity": 4
}

```

**Reply by lddlww** [reply]
i think i find some problem， it separte 4 pools in minio
but i used the follow command to start minio：
```
ExecStart=/usr/local/bin/minio server http://10.5.17.97:9022/data/minio{4...27} \
                                      http://10.5.17.98:9022/data/minio{4...27} \
                                      http://10.5.17.99:9022/data/minio{4...27} \
                                      http://10.5.17.100:9022/data/minio{4...27} \
                                        --address 10.5.17.97:9022 --console-address 10.5.17.97:9023
```

**Reply by lddlww** [reply]
oh ，sorry i am mistake

**Comment by klauspost** [comment, marked as answer]
@lddlww Your servers are configured as independent server pools. So there is no redundancy between servers.

https://docs.min.io/minio/baremetal/installation/deployment-and-management.html

**Reply by lddlww** [reply]
@klauspost thank you very much
